packages/story-editor/src/app/media/utils/useMediaUploadQueue/reducer.js

```diff
@@ -63,14 +63,16 @@ export function addItem(
   }
 ) {
   const id = uuidv4();
+
+  if (!resource.id) {
```

Comment:
What if resource is null?

Reply:
Then this should still kick in and add an ID. That's intended.

Reply:
Jonny was asking about the case when `resource` itself is `null` -- then there wouldn't be anything to add the ID to.

Reply:
Ah, got it.

The `uploadMedia` callback in `useUploadMedia` always passes a `resource` to `addItem()`. It's expected to never be null:

https://github.com/google/web-stories-wp/blob/4c23ee8ef2d5b38185d7f974751401987df5e367/packages/story-editor/src/app/media/useUploadMedia.js#L287-L296

Reply:
Is it worth checking it is an object anyway, just to be careful?

Reply:
It's never not an object.